Directory.Build.props

```diff
@@ -29,6 +29,7 @@
     <BaseOutputPath Condition="'$(BaseOutputPath)'==''">$(BinDir)</BaseOutputPath>
     <ObjDir Condition="'$(ObjDir)'==''">$(BinDir)obj/</ObjDir>
     <BaseIntermediateOutputPath Condition="'$(BaseIntermediateOutputPath)'==''">$(ObjDir)</BaseIntermediateOutputPath>
+    <RootIntermediateOutputPath Condition="'$(RootIntermediateOutputPath)'==''">$(ObjDir)</RootIntermediateOutputPath>
```

Comment:
Can we delete BaseInterpediateOutputPath and replace its usages with RootIntermediateOutputPath?

Reply:
`BaseIntermediateOutputPath` is the actual path used by MSBuild, it would probably be better to get rid of `RootIntermediateOutputPath` instead.

Reply:
With old buildtools `BaseIntermediateOutputPath` used to mean `$RepoRoot\bin\obj`.  `RootIntermediateOutputPath` is the NEW buildtools variable for this folder.

I can remove setting `BaseIntermediateOutputPath` this high up, and only set it here:

https://github.com/dotnet/machinelearning/blob/98ac03b7e3210fab970b6983aee7f1fd4e021b7d/Directory.Build.props#L101-L108

Reply:
Talked with @eerhardt offline and I better understand where we are now and why, and where we want to be.

Reply:
Ok, I think I left this code in a better place.  Let me know what you think now.